modules/cert-manager/provider.tf

```diff
@@ -9,3 +9,7 @@ terraform {
 provider "kubectl" {
   config_path = pathexpand("${var.k8s_config_path}")
 }
+
+provider "kubernetes" {
```

Comment:
move providers to parent (the project which use this module) and just pass them by using providers mapping
have provider requirement.tf file with content like this:
terraform {
  required_providers {
    kubectl = {
      source                = "???"
    }
    kubernetes = {
      source                = "???"
    }
  }
}